### PR TITLE
feat: handle links in same window

### PR DESCRIPTION
**Pull request by goldjacobe**
**Context:**

**Changes In This Pull Request:**

**Test Plan:**

### CODE REVIEW

**Comment by goldjacobe** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2** <a href="https://app.graphite.dev/github/pr/withgraphite/graphite-desktop/2" target="_blank">Graphite</a> 
    * **PR #3** <a href="https://app.graphite.dev/github/pr/withgraphite/graphite-desktop/3" target="_blank">Graphite</a> 
      * **PR #4** <a href="https://app.graphite.dev/github/pr/withgraphite/graphite-desktop/4" target="_blank">Graphite</a> 
        * **PR #5** <a href="https://app.graphite.dev/github/pr/withgraphite/graphite-desktop/5" target="_blank">Graphite</a>  👈
          * **PR #6** <a href="https://app.graphite.dev/github/pr/withgraphite/graphite-desktop/6" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/withgraphite/graphite-desktop/5?utm_source=stack-comment).

File: src/index.ts [hunk 1]
```diff
@@ -26,6 +26,13 @@ function createWindow() {
     return { action: 'deny' }
   })
 
+  win.webContents.on('will-navigate', (event, url) => {
+    if(!url.startsWith('https://app.graphite.dev')) {
+      event.preventDefault();
```
Comment (nicholasyan) [inline src/index.ts:31]: What is the default behavior we're stopping here? (I'm guessing opening a new window?)

File: src/index.ts [hunk 2]
```diff
@@ -26,6 +26,13 @@ function createWindow() {
     return { action: 'deny' }
   })
 
+  win.webContents.on('will-navigate', (event, url) => {
+    if(!url.startsWith('https://app.graphite.dev')) {
```
Comment (nicholasyan) [inline src/index.ts]: Why do we care whether this is a Graphite URL or a non-Graphite URL? I'd expect that we'd want all links to open in the same window?

File: src/index.ts [hunk 3]
```diff
@@ -26,6 +26,13 @@ function createWindow() {
     return { action: 'deny' }
   })
 
+  win.webContents.on('will-navigate', (event, url) => {
```
Comment (nicholasyan) [inline src/index.ts:29]: If the motivation is handling links in the same window, would the above `setWindowOpenHandler` solve this?

**Review by nicholasyan — APPROVED** [review]
I trust you since this is small/just for testing, but some context would've been helpful in the description of this PR

File: src/index.ts [hunk 4]
```diff
@@ -26,6 +26,13 @@ function createWindow() {
     return { action: 'deny' };
```
Comment (goldjacobe) [inline src/index.ts:26]: <i class='graphite__hidden'>[Re: line 24]<br/></i>

this is only for links opened in new windows<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/withgraphite/graphite-desktop/5?utm_source=unchanged-line-comment">Graphite</a>.</span>

File: src/index.ts (hunk 3 above)
Comment (goldjacobe) [reply src/index.ts:29]: that one is for links that don't open a new window (e.g. from PR comments)

File: src/index.ts (hunk 1 above)
Comment (goldjacobe) [reply src/index.ts:31]: prevent us from navigating to the link

**Comment by goldjacobe** [comment]
**Graphite Merge Job**

Current status: ✅ Merged

This pull request was successfully merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/withgraphite/graphite-desktop/5?utm_source=stack-comment).


Job Reference: XR3SHLJDtJFBM9xieqeZ

**Pull request merged**